sound/soc/sof/intel/hda-dsp.c

```diff
@@ -452,6 +452,48 @@ static int hda_dsp_set_D0_state(struct snd_sof_dev *sdev,
 	return ret;
 }
 
+/* helper to log DSP state */
+static void hda_dsp_state_log(struct snd_sof_dev *sdev)
+{
+	switch (sdev->dsp_power_state.state) {
+	case SOF_DSP_PM_D0:
+		switch (sdev->dsp_power_state.substate) {
+		case SOF_HDA_DSP_PM_D0I0:
+			dev_dbg(sdev->dev, "Current DSP power state: D0I0\n");
+			break;
+		case SOF_HDA_DSP_PM_D0I3:
+			dev_dbg(sdev->dev, "Current DSP power state: D0I3\n");
+			break;
+		default:
+			dev_dbg(sdev->dev,
+				"Unknown DSP power state: %d, substate: %d\n",
+				sdev->dsp_power_state.state,
+				sdev->dsp_power_state.substate);
```

Comment:
Verrrrrry minor nitpick: You could say that it's an unknown substate of D0.

Reply:
fixed @paulstelian97